backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -489,18 +530,31 @@ boolean childContains(Set<?> list, Set<?> sublist) {
     CommitEntity internalCommitA;
     CommitEntity internalCommitB;
     try (Session readSession = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      RepositoryEntity repositoryEntity = repositoryFunction.apply(readSession);
+      RepositoryEntity repositoryEntity =
+          repositoryDAO.getRepositoryById(readSession, request.getRepositoryId());
 
-      internalCommitA = readSession.get(CommitEntity.class, request.getCommitShaA());
-      if (internalCommitA == null) {
-        throw new ModelDBException(
-            "No such commit found : " + request.getCommitShaA(), Status.Code.NOT_FOUND);
-      }
+      if (request.getBranchA().isEmpty() || request.getBranchB().isEmpty()) {
```

Comment:
same as commit

Reply:
Resolved as per attached scenarios image on the description